File: app/src/main/java/com/bsstokes/bsdiy/db/sqlite/migrations/Migrator.java
package com.bsstokes.bsdiy.db.sqlite.migrations;

import android.database.sqlite.SQLiteDatabase;

public class Migrator {

    private final Migration[] migrations;

    public Migrator(Migration[] migrations) {
        this.migrations = migrations;
    }

    public void migrate(SQLiteDatabase database, int oldVersion, int newVersion) {
        for (int version = oldVersion; version < newVersion; ++version) {
            migrations[version].onUpgrade(database);
        }
    }

    public int getVersion() {
        return migrations.length;
    }
}

File: app/src/main/java/com/bsstokes/bsdiy/db/sqlite/migrations/Migrator.kt
package com.bsstokes.bsdiy.db.sqlite.migrations

import android.database.sqlite.SQLiteDatabase

class Migrator(private val migrations: Array<Migration>) {

    fun migrate(database: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        for (version in oldVersion..newVersion - 1) {
            migrations[version].onUpgrade(database)
        }
    }

    val version: Int
        get() = migrations.size
}
